[16.01] Add a requirements file suitable for use with `conda {create,install} --file`

natefoo opened the pull request:
Should ease the dependency transition for Conda users.

natefoo committed:
Add a requirements file suitable for use with `conda {create,install} --file`.

jmchilton commented on lib/galaxy/dependencies/conda-requirements.txt:
```diff
@@ -0,0 +1,63 @@
+# newer release.
+#
+# https://bitbucket.org/mchaput/whoosh/pull-requests/50/fix-lru-thread-safety/diff
+#Whoosh==2.4.1+gx1
```
Can you place a URL to our change in here for these things then?

natefoo replied:
There's sorta not... it's a sed one-liner added by Starforge when it's built: https://github.com/galaxyproject/starforge/blob/master/wheels/build/wheels.yml#L171

jmchilton replied:
You can link to the wheel directly in this file right - add the line:

```
https://wheels.galaxyproject.org/packages/Whoosh-2.4.1+gx1-py2-none-any.whl
```

Or not?

jmchilton replied:
Looks like it is not possible - can do it with `conda-env` though if we want to support that someday https://github.com/conda/conda-env/.

jmchilton replied:
Here is what I would do - rename this to conda-environment.txt since it isn't a requirements.txt file and augment with a second `conda-requirements.txt` file that has the wheel locations (for whoosh, pysam, etc...).

```
conda create galaxy --file conda-environment.txt
source activate galaxy
pip install -r conda-requirements.txt
```

natefoo replied:
I didn't create a separate `conda-requirements.txt` since `requirements.txt` can already be used for this purpose.

natefoo commented:
Worth mentioning here that that plan to move wheels.yml over from Starforge and using it to generate all these files grows in importance as we keep adding requirements files.

natefoo committed:
Rename conda-requirements.txt to conda-environment.txt since it is not a
requirements file, include instructions on how to install remaining
dependencies from pip.
